ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/config/MergedApplicationConfig.kt

```diff
@@ -8,9 +8,18 @@ package io.ktor.server.config
  * Merge configuration combining all their keys.
  * If key is not found in one of the configs, search will continue in the next config in the list.
  */
+@Deprecated("Use mergeWith/mergeWithFallback instead.")
 public fun List<ApplicationConfig>.merge(): ApplicationConfig {
     require(isNotEmpty()) { "List of configs can not be empty" }
-    return foldRight(last()) { config, acc -> MergedApplicationConfig(config, acc) }
+    return foldRight(last()) { config, acc -> config.mergeWithFallback(acc) }
+}
+
+public fun ApplicationConfig.mergeWith(other: ApplicationConfig): ApplicationConfig {
+    return MergedApplicationConfig(other, this)
+}
+
+public fun ApplicationConfig.mergeWithFallback(other: ApplicationConfig): ApplicationConfig {
```

Comment:
`mergeWithFallback` -> `withFallback`

Reply:
I think it should be ok. Also, it mimics one of the libraries that we use for configs
https://lightbend.github.io/config/latest/api/com/typesafe/config/Config.html#withFallback-com.typesafe.config.ConfigMergeable-